### DISCUSSION TITLE
List all PWAs with their IDs in terminal?

### DISCUSSION THREAD

**Discussion by markcellus** [filips123/PWAsForFirefox#169] [Q&A]
Hello!

Is there any way we can do `firefoxpwa site list` similar to `firefoxpwa profile list`, so that I can see all PWAs installed? There seems to be an issue where there are two of the same PWA .desktop files are installed, but it isn't reflected in the extension UI. A `firefoxpwa site list` would allow me to find the IDs of the desktop files that I need to remove via `firefoxpwa site uninstall <ID>`

Thanks

**Comment by filips123** [comment]
Why don't you just use `firefoxpwa profile list`? It already lists PWA IDs (in parentheses), and .desktop filenames are `FFPWA-ID.desktop`.

**Reply by markcellus** [reply]
I used that. But for some reason, there are still PWA sites installed that don't appear on that list. I don't know how this is the case though but it's happened quite a few times already after the different version upgrades. I had to manually find and remove the .desktop files which has temporarily fixed the issue. But it's likely to happen again. 

So, just wondering if there is a way we can have a new `firefoxpwa site list` that will list all of firefoxpwa's .desktop files for apps, regardless of whether they appear on the `firefoxpwa profile list` or not.

**Reply by filips123** [reply]
Well, that seems like a bug. `firefoxpwa profile list` lists all PWAs from `config.json`, so it appears that it was missing some PWAs for some reason. But in this case, `firefoxpwa site uninstall` shouldn't work either as it checks the config file if the site exists.

How were that duplicated .desktop files named and when exactly did they appear?

`firefoxpwa site list` would be harder to implement as it would need to support more than just Linux and other OSes use different ways to store apps. And even then, that would just be some workaround for the actual problem.

**Reply by markcellus** [reply]
`firefoxpwa site uninstall` requires the ID of the pwa, which is exactly what I had to manually hunt down :smile: . The ID seems to be prefixed in the .desktop file names. And I had to open each of my .desktop files to see what PWAs where before uninstalling. It would have been better if I could just run `firefoxpwa site list` to get the IDs and verify they're the apps that I actually want to delete.

**Reply by filips123** [reply]
My point is that it should never happen that some PWA exists but is not shown in `firefoxpwa profile list`, as that would mean PWA didn't get added to a profile. So, when this happens next time, please show your `config.json` and logs so I can check what caused this.
